src/main/scala/org/broadinstitute/dsde/firecloud/utils/StreamingActor.scala

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.dsde.firecloud.utils
+
+import akka.actor.{Actor, ActorContext}
+import org.slf4j.{Logger, LoggerFactory}
+import spray.http._
+import spray.routing.{HttpService, RequestContext}
+
+object StreamingActor {
+  case class FirstChunk(httpData: HttpData, remaining: Int)
+  case class NextChunk(httpData: HttpData, remaining: Int)
+  case class Ok(remaining: Int)
+}
+
+class StreamingActor(ctx: RequestContext, contentType: ContentType, contentDisposition: String) extends Actor with HttpService {
```

Comment:
I find the `contentDisposition` variable name confusing here - the value is actually the attachment filename. (I know it goes inside the content-disposition header, but it's still the filename)